### ISSUE TITLE
[Proposal] Flogo App Api

### ISSUE THREAD

**Issue by fm-tibco** [TIBCOSoftware/flogo#289]
**I'm submitting a ...**  (check one with "x")
```
[] bug report => search github for a similar issue or PR before submitting
[x] feature request
[] support request
[] general question
```
Proposal to create a simplified Go developer API to create flogo apps.  Please provide any feedback or suggestion using this github issue.

Wiki:
https://github.com/TIBCOSoftware/flogo/wiki/Go-App-Api-Proposal

Branch:
https://github.com/TIBCOSoftware/flogo-lib/tree/go_api

**Comment by retgits** [comment]
I really like this proposal! Will we be able to reuse activities in this scenario as well? So for example, looking at the `Simple Application using REST trigger and function handler`

```go
func main() {

	app := flogo.NewApp()

	trg := app.NewTrigger(&rest.RestTrigger{}, map[string]interface{}{"port": 8080})
	trg.NewFuncHandler(map[string]interface{}{"method": "GET", "path": "/blah"}, LogMessage)

	e, err := flogo.NewEngine(app)

	if err != nil {
		logger.Error(err)
		return
	}

	engine.RunEngine(e)
}

func LogMessage(ctx context.Context, inputs map[string]*data.Attribute) (map[string]*data.Attribute, error) {
	logger.Infof("#v", inputs)
	return nil, nil
}
```

Could I use the [Log activity](https://github.com/TIBCOSoftware/flogo-contrib/tree/master/activity/log) instead of the LogMessage function?

**Comment by mellistibco** [comment]
Closing as the initial proposal has been merged into master and tagged as part of 5.5
